pkgs/by-name/he/heimdall/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  cmake,
+  enableGUI ? false,
+  fetchFromSourcehut,
+  gitUpdater,
+  lib,
+  libusb1,
+  pkg-config,
+  qt6,
+  stdenv,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "heimdall";
+  version = "2.2.1";
+
+  src = fetchFromSourcehut {
+    owner = "~grimler";
+    repo = "Heimdall";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-x+mDTT+oUJ4ffZOmn+UDk3+YE5IevXM8jSxLKhGxXSM=";
+  };
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ] ++ lib.optional enableGUI qt6.wrapQtAppsHook;
+
+  buildInputs = [
+    (libusb1.override { withStatic = stdenv.hostPlatform.isWindows; })
+  ] ++ lib.optional enableGUI qt6.qtbase;
+
+  postFixup = lib.optional enableGUI ''
+    # heimdall cli looked up from PATH by gui
+    wrapProgram $out/bin/heimdall-frontend \
+      --prefix PATH : $out/bin
+  '';
```

Comment:
Are we sure this is required on v2.2.1? The GUI appears to work just fine for me without this.

Reply:
It won't work if you try to run it using `nix run` without having heimdall installed or having it inside a nix shell.
I mean the gui will launch, but try for example printing pit and it'll fail.

unless the upstream added another way of finding `heimdall` binary in 2.2.1

Reply:
I would prefer simply patching in the full path to the binaries rather than wrapping with all its uglyness.

Reply:
idk this seems simpler and I can't imagine any issues with just wrapping it, lol

Reply:
To be fair, this is now a double-wrap, once for PATH and once for Qt. That means the following files now exist, which at least I can't immediately understand which is which (although file sizes do make it obvious when looking closer):

```
$ ls -al result/bin/ | grep frontend
-r-xr-xr-x 1 root root  16080  1. Jan 1970  heimdall-frontend
-r-xr-xr-x 1 root root 306328  1. Jan 1970  .heimdall-frontend-wrapped
-r-xr-xr-x 1 root root  15928  1. Jan 1970  .heimdall-frontend-wrapped_
```

Reply:
fair enough